engines/router/missionctl/server/application.go

```diff
@@ -67,26 +67,31 @@ func Run() {
 		}
 
 		s := upi.NewUPIServer(missionCtl)
-		if cfg.AppConfig.CustomMetrics {
-			m := cmux.New(l)
-			grpcL := m.Match(cmux.HTTP2HeaderField("content-type", "application/grpc"))
-			httpL := m.Match(cmux.HTTP1Fast())
+		m := cmux.New(l)
+		grpcL := m.Match(cmux.HTTP2HeaderField("content-type", "application/grpc"))
+		httpL := m.Match(cmux.Any())
 
-			mux := http.NewServeMux()
+		mux := http.NewServeMux()
+		mux.Handle("/v1/internal/", http.StripPrefix(
+			"/v1/internal",
+			handlers.NewInternalAPIHandler([]string{
+				cfg.EnsemblerConfig.Endpoint,
+				cfg.EnrichmentConfig.Endpoint,
+			}),
+		))
+		if cfg.AppConfig.CustomMetrics {
 			mux.Handle("/metrics", promhttp.Handler())
-			httpS := &http.Server{Handler: mux}
-
-			go s.Run(grpcL)
-			go func() {
-				if err := httpS.Serve(httpL); err != nil {
-					log.Glob().Errorf("Failed to serve http server: %s", err)
-				}
-			}()
-			if err := m.Serve(); err != nil {
-				log.Glob().Errorf("Failed to serve cmux: %s", err)
+		}
+		httpS := &http.Server{Handler: mux}
```

Comment:
Super nit: I was wondering why this is `https`. 😅  May be `httpServer` would be clearer?

Reply:
oops, copied it from the mux readme, will do so 👍